### PR TITLE
Upgrade LLVM to 18.1.8

### PR DESCRIPTION
**Pull request by Databean**
Bug: b/385208338

### CODE REVIEW

**Review by ser-io — APPROVED** [review]

**Comment by Databean** [comment]
Changed PR to be from LLVM 19.1.0 to LLM 18.1.8, it seems the latest one requires a glibc that is too new.

File: base/cvd/cuttlefish/common/libs/utils/files.cpp [hunk 1]
```diff
@@ -536,13 +536,17 @@ Result<std::string> ReadFileContents(const std::string& filepath) {
...
+    if (errno == ERANGE) {
+      process_wd.resize(process_wd.size() * 2, ' ');
+    } else {
+      PLOG(ERROR) << "getcwd failed";
+      return "";
+    }
   }
-  return process_cwd;
+  // Will find the null terminator and size the string appropriately.
+  return std::string(process_wd.data());
```
Comment (jemoreira) [inline base/cvd/cuttlefish/common/libs/utils/files.cpp:549]: nit: `return process_wd.data()` is enough

**Review by jemoreira — APPROVED** [review]

File: base/cvd/cuttlefish/common/libs/utils/files.cpp (hunk 1 above)
Comment (Databean) [reply base/cvd/cuttlefish/common/libs/utils/files.cpp]: Done.

**Comment by Databean** [comment]
Thanks for reviewing!

**Pull request merged**